rdt/transformers/pii/anonymizer.py

```diff
@@ -126,6 +124,7 @@ def _fit(self, data):
             data (pandas.Series):
                 Data to fit to.
         """
+        self.faker.seed_instance(self._random_seed)
```

Comment:
This is the main change: we set the seed during `fit` instead of during initialization.